### PR TITLE
Add header to MIT License

### PR DESCRIPTION
**Pull request by sandakersmann**
Add header to MIT License

### CODE REVIEW

File: LICENSE-MIT [hunk 1]
```diff
@@ -1,3 +1,5 @@
+Add header to MIT License
```
Comment (teor2345) [inline LICENSE-MIT]: This looks like a note about making change. Did you mean to write the actual change here?

**Review by teor2345 — DISMISSED** [review]
Thanks, but it seems like this change is incomplete?

I'm also not sure if we're looking for help maintaining our legal documents like licenses, because changes need to go through legal review. Our other documentation is here, it is a lot more flexible:
https://github.com/ZcashFoundation/zebra/tree/main/book

File: LICENSE-MIT (hunk 1 above)
Comment (sandakersmann) [reply LICENSE-MIT]: Went a bit fast :D Fixed now

**Review by teor2345 — COMMENTED** [review]
Here's the automated check that GitHub uses:
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository#detecting-a-license

```sh
$  licensee detect
License:        NOASSERTION
Matched files:  LICENSE-APACHE, LICENSE-MIT, README.md
...
LICENSE-MIT:
  Content hash:  4c2c763d64bbc7ef2e58b0ec6d06d90cee9755c9
  Attribution:   Copyright (c) 2019-2023 Zcash Foundation
  Confidence:    100.00%
  Matcher:       Licensee::Matchers::Exact
  License:       MIT
...
```

Which is the same as the `main` branch:
```sh
$ licensee detect
License:        NOASSERTION
Matched files:  LICENSE-APACHE, LICENSE-MIT, README.md
...
LICENSE-MIT:
  Content hash:  4c2c763d64bbc7ef2e58b0ec6d06d90cee9755c9
  Attribution:   Copyright (c) 2019-2023 Zcash Foundation
  Confidence:    100.00%
  Matcher:       Licensee::Matchers::Exact
  License:       MIT
...
```

File: LICENSE-MIT [hunk 2]
```diff
@@ -1,3 +1,5 @@
+The MIT License (MIT)
```
Comment (teor2345) [inline LICENSE-MIT:1]: Just double-checking this is ok with people who understand legal stuff

Comment (oxarbitrage) [reply LICENSE-MIT:1]: I don't understand legal stuff but i think this is not needed:

https://opensource.org/license/mit/
https://en.wikipedia.org/wiki/MIT_License

**Comment by oxarbitrage** [comment]
We appreciate the contribution but we are going to prefer to keep licenses as they are unless changes are really needed. Thanks and feel free to push changes in other parts of Zebra.

**Pull request closed without merging**